code/modules/mob/living/living_defense.dm

```diff
@@ -68,6 +68,19 @@
 		else
 				return 0
 
+/mob/living/proc/set_combat_mode(new_mode, silent)
+	if(combat_mode == new_mode)
+		return
+	combat_mode = new_mode
```

Comment:
```suggestion
	if(combat_mode == new_mode)
		return
	. = combat_mode
	combat_mode = new_mode
```
This works nice for redefinitions that call parent (ideally they all do), to know if a change has happened, with `isnull(.)`